airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml

```diff
@@ -0,0 +1,404 @@
+version: "0.1.0"
+
+definitions:
+  schema_loader:
+    type: JsonSchema
+    file_path: "./source_greenhouse/schemas/{{ options['name'] }}.json"
+  selector:
+    type: RecordSelector
+    extractor:
+      type: DpathExtractor
+      field_pointer: []
+  requester:
+    type: HttpRequester
+    name: "{{ options['name'] }}"
+    url_base: "https://harvest.greenhouse.io/v1/"
+    http_method: "GET"
+    authenticator:
+      type: BasicHttpAuthenticator
+      username: "{{ config['api_key'] }}"
+  retriever:
+    type: SimpleRetriever
+    name: "{{ options['name'] }}"
+    primary_key: "{{ options['primary_key'] }}"
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: LimitPaginator
+      page_size: 100
+      pagination_strategy:
+        type: "CursorPagination"
+        cursor_value: "{{ headers['link']['next']['url'] }}"
+        stop_condition: "{{ 'next' not in headers['link'] }}"
+      limit_option:
+        field_name: "per_page"
+        inject_into: "request_parameter"
+      page_token_option:
+        inject_into: "path"
+      url_base: "*ref(definitions.requester.url_base)"
+  applications_stream:
+    type: DeclarativeStream
```

Comment:
do we need `type: DeclarativeStream`? can't we use a default impl since this is probably the only type of stream we'll have for a while?